.travis.yml

```diff
@@ -1,41 +1,160 @@
-os:
-  - linux
-  - osx
+#  2020-05-26: NOTE: Enabling experimental Travis feature of YAML check inside
+# Look into: Build #NUM -> Job #NUM.N -> View config -> Build config validation
+version: ~> 1.0
+
+# NOTE: Please, be aware that Travis YAML & docs & API are hard to make work properly. Travis configuration requires a lot of retries and some compromises. Tere are many ways that may look like that can be done in that way, but it would not work most of the time, or not work the way that you expect, need it. Travis config works only the certain particular ways. Some things look possible - but they are impossible in Travis. Current configuration is "the best way possible" that was found in ~100-150-200 retries, depending on what concider a retry.
+
+# NOTE: Let the official Travis YAML checker help you: https://config.travis-ci.com/explore
 
 language: nix
 
-sudo: required
+sudo: required    #  2020-05-26: NOTE: Despite deprecated, but siletly still RAM 4GB -> 7.5GB
 
 git:
+  quiet: true # NOTE: Do not log
   depth: 4    # NOTE: "The use of clone depth: 1 often results in a git error
               # when a new commit has been pushed to a branch before the CI
               # platform started cloning the intended commit."
 
+# NOTE: {os} x {jobs} - {exclude} = {build matrix}
+os:
+  - linux
+  - osx
+
+#  2020-05-26: NOTE: Currently newest Travis dist Ubuntu 18.04 bionic
+dist: bionic
+
+# #  2020-05-26: NOTE: Currently newest macOS image
+# osx_image: xcode11.4  # NOTE: Official Nix installer fails spectacularly on it.
```

Comment:
It is Nix upstream issues.
I wound not report the installer bugs there.
I would not fix the upstream Nix installer, or contribute code into it.

Since I already did all those reports and major installer work 3 years ago, and I wait 3 years on some response/collaboration from them.

Since I know it is useless to send their Reports or the code - I see no point sending them there.

Then also I am not only helping them kick the installer mess can down the road, I would also work against my own interest and major work I've already done for them

Reply:
So that is not a TODO.
It is upstream installer bugs.
Knowing that it may never resolve - I've just put date there.

Ther is no rush to TODO it.
If someone would need to test on newer macOS - they would read the date, then try to enable that option and be able to test if it works - that is all.

We probably would migrate to the other CI with other macOS images before the Nix installer resolves its issues.